sdk/core/azure-core/inc/azure/core/internal/deferred_operation.hpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include <exception>
+#include <functional>
+
+namespace Azure { namespace Core {
+  template <typename T> class DeferredResponse;
+
+  namespace _internal {
+
+    /**
+     * @brief Base class for deferred operation shared state. Wraps the outgoing request.
+     *
+     * This class exists to allow DeferredResponseSharedWithCallback objects to be aggregated
+     * where the specialization of the DeferredResponseSharedWithCallback has different
+     * specialization types.
+     *
+     */
+    class DeferredResponseSharedBase {
+      friend class DeferredResponseFactory;
+
+    public:
+      Azure::Core::Http::Request m_request;
+
+      virtual Azure::Core::Http::Request Request() { return m_request; }
+      /**
+       * @brief Process a raw response received from the service and save it for later
+       * retrieval.
+       *
+       * Called from the DeferredResponseProcessor when processing deferred operations.
+       *
+       * @param response Response returned by the service.
+       */
+      virtual void ProcessRawResponse(std::unique_ptr<Azure::Core::Http::RawResponse>& response)
+          = 0;
+
+      /**
+       * @brief Creates a new DeferredResponseSharedBase for the specified request.
+       *
+       * @param request Request to send to the server eventually.
+       */
+      explicit DeferredResponseSharedBase(Azure::Core::Http::Request request) : m_request(request)
```

Comment:
See my other comment. Storage code doesn't have access to `Azure::Core::Http::Request`.